e2e/cypress/integration/enterprise/system_console/groups_assign_roles_spec.js

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. # Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+// # Function to get all the teams associated to group and unlink them
+const getTeamsAssociatedToGroupAndUnlink = (groupId) => {
+    cy.request({
+        headers: {'X-Requested-With': 'XMLHttpRequest'},
+        url: `/api/v4/groups/${groupId}/teams`,
+        method: 'GET',
+    }).then((response) => {
+        expect(response.status).to.equal(200);
+        response.body.forEach((element) => {
+            cy.request({
+                headers: {'X-Requested-With': 'XMLHttpRequest'},
+                url: `/api/v4/groups/${element.group_id}/teams/${element.team_id}/link`,
+                method: 'DELETE',
+            });
+        });
+    });
+};
+
+// # Function to get all the channels associated to group and unlink them
+const getChannelsAssociatedToGroupAndUnlink = (groupId) => {
+    cy.request({
+        headers: {'X-Requested-With': 'XMLHttpRequest'},
+        url: `/api/v4/groups/${groupId}/channels`,
+        method: 'GET',
+    }).then((response) => {
+        expect(response.status).to.equal(200);
+        response.body.forEach((element) => {
+            cy.request({
+                headers: {'X-Requested-With': 'XMLHttpRequest'},
+                url: `/api/v4/groups/${element.group_id}/channels/${element.channel_id}/link`,
+                method: 'DELETE',
+            });
+        });
+    });
+};
+
+describe('System Console', () => {
+    it('MM-20058 - System Admin can map roles to teams and channels via group configuration page', () => {
+        cy.apiLogin('sysadmin');
+
+        // # Go to system admin page and to team configuration page of channel "eligendi"
+        cy.visit('/admin_console/user_management/groups');
+        cy.get('#developers_group').then((el) => {
+            if (el.text().includes('Edit')) {
+                cy.get('#developers_edit').then((buttonEl) => {
+                    // # Get the Group ID and remove all the teams and channels currently attached to it then click the button
+                    const groupId = buttonEl[0].href.match(/\/(?:.(?!\/))+$/)[0].substring(1);
+                    getTeamsAssociatedToGroupAndUnlink(groupId);
+                    getChannelsAssociatedToGroupAndUnlink(groupId);
+                    cy.get('#developers_edit').click();
+                });
+            } else {
```

Comment:
Can you also add a step to click on "Not Linked" link, if it's present. This way, it will also handle the case where the group wasn't linked before. You can do this by adding the following step in the else condition before clicking on configure button. 

```
                // # Link the Group if its not linked before
                if (el.find('.icon.fa-unlink').length > 0) {
                    el.find('.icon.fa-unlink').click();
                }
```